### PR TITLE
Fuzzy search for tags

### PR DESCRIPTION
**Pull request by martinraison**
@stkem @2is10

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular-black/src/tags/tagService.js [hunk 1]
```diff
@@ -97,20 +97,15 @@ angular.module('kifi.tagService', [
...
-            newList.push(tag);
-            ins++;
-          }
+        var newList = allTags;
+        if (term.length) {
+          var options = {
+            keys: ['name'],
+            threshold: 0.3
+          };
+          var f = new Fuse(allTags, options);
```
Comment (stkem) [inline kifi-backend/modules/shoebox/angular-black/src/tags/tagService.js]: perhaps move this to a higher level (so it't not reconstructed on every search) if possible?

**Comment by stkem** [comment]
made one small optimization comment, lgtm otherwise

File: kifi-backend/modules/shoebox/angular-black/src/tags/tagService.js (hunk 1 above)
Comment (martinraison) [inline kifi-backend/modules/shoebox/angular-black/src/tags/tagService.js]: good point - it used to be necessary to recreate the Fuse object for every search, but it looks like it's been fixed.

**Comment by 2is10** [comment]
cool. we’ll probably want to use the same library in the extension.

**Pull request merged**